### ISSUE TITLE
Prefer using null-aware operators

### ISSUE THREAD

**Issue by hterkelsen** [dart-lang/linter#159]
BAD:

``` dart
return foo != null ? foo : const [];

if (foo == null) foo = new Set();

if (foo != null) foo.bar();
```

GOOD:

``` dart
return foo ?? const [];

foo ??= new Set();

foo?.bar();
```

**Comment by a14n** [comment]
This issue can be closed as we now have `prefer_null_aware_operators`, `prefer_if_null_operators` and `prefer_conditional_assignment`.

**Comment by pq** [comment]
Super.  Thanks!
